### PR TITLE
Add stars metric

### PR DESCRIPTION
**Pull request by samuelli**
This adds a stars metric which gets the raw star data to allow for star trends to be mapped (not currently exposed yet).

### CODE REVIEW

**Comment by samuelli** [comment]
This could be implemented in a different way to fetch total stars much more efficiently, but it would not allow us to analyze stars over time, which would be useful to correlate metrics.

File: src/metrics/stars.ts [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+  constructor(stars: Star[]) {
+    this.stars = stars;
+  }
+
+  summary(): string {
+    return `There are ${this.stars.length} stars.`;
+  }
+}
+
+type StarOpts = {
```
Comment (aomarks) [inline src/metrics/stars.ts]: `StarsOpts` I think

File: src/metrics/stars.ts [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+  const results =
+      github.cursorQuery<StarsQuery, StarsQueryVariables>(
+          starsQuery,
+          {owner, name},
+          (data) => data.repository && data.repository.stargazers);
+
+  for await (const data of results) {
+    const repo = data.repository;
+    if (!repo) {
+      break;
```
Comment (aomarks) [inline src/metrics/stars.ts]: `break` or `continue`?

File: src/test/metrics/stars-test.ts [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+  const {server, client} = await startTestReplayServer(t);
+  t.context.server = server;
+  t.context.client = client;
+});
+
+test.afterEach.cb((t) => {
+  t.context.server.close(t.end);
+});
+
+test('WebComponents stars', async (t) => {
```
Comment (aomarks) [inline src/test/metrics/stars-test.ts]: I think you need to `git add` the test replay files.

Comment (aomarks) [reply src/test/metrics/stars-test.ts]: Oh sorry, just saw your comment. SG.

Comment (samuelli) [reply src/test/metrics/stars-test.ts]: Done

File: src/metrics/stars.ts (hunk 1 above)
Comment (samuelli) [reply src/metrics/stars.ts]: Done.

File: src/metrics/stars.ts (hunk 2 above)
Comment (samuelli) [reply src/metrics/stars.ts]: `continue`. Searching, there's a few other instances where this looks like its probably wrong. Going to send a separate PR to fix.

**Pull request merged**